### DISCUSSION TITLE
Question About Deprecated Property in spring-boot-starter-data-aerospike:0.15.0

### DISCUSSION THREAD

**Discussion by flagmen** [aerospike/spring-data-aerospike-starters#255] [Q&A] [answered]
Hello, Dear community.

We are using the following versions in our project:
  - spring-boot-starter-data-aerospike: 0.15.0
  - spring-data-aerospike: 4.8.0
  - spring-boot-autoconfigure-data-aerospike: 0.15.0

We noticed that the property `spring.aerospike.hosts` is marked as deprecated. However, the `AerospikeAutoConfiguration` class still requires this property to work, as shown by the `@ConditionalOnProperty("spring.aerospike.hosts")` annotation.
We want to configure `ClientPolicy` without using deprecated properties. Right now, the only way to do this is by extending `AbstractAerospikeDataConfiguration` and manually defining ClientPolicy.

In the past, we used _spring-boot-starter-data-aerospike version 0.12.0_, where we could set this in the `application.yml` without any extra steps.

Is there a way to configure `ClientPolicy` in the current version without using deprecated properties and without manual configuration?

Thanks for your help.

**Comment by agrgr** [comment]
Hello @flagmen!

Thanks for the question.
Marking certain configuration properties as deprecated was done before the actual changes in the next release of the starters project. Currently there is no way to configure `ClientPolicy` without using deprecated properties. In the next release it is planned to replace some of the prefixes used in application.properties (including hosts and namespace) with the ones directly supported by Spring Data Aerospike.

**Reply by agrgr** [reply, marked as answer]
Hello again @flagmen,

Please note that the next version 0.16 is released. Within this version the deprecation of the properties is rolled back after getting feedback about necessity to keep properties names the same.
Besides, the newer version contains an update to allow initializing only client beans, and an addition of durableDelete property for batch ClientPolicies.
